crates/utils/src/settings/structs.rs

```diff
@@ -1,68 +1,65 @@
-use merge::Merge;
 use serde::Deserialize;
-use std::net::IpAddr;
+use std::net::{IpAddr, Ipv4Addr};
 
-#[derive(Debug, Deserialize, Clone, Merge)]
+#[derive(Debug, Deserialize, Clone, SmartDefault)]
+#[serde(default)]
 pub struct Settings {
-  pub(crate) database: Option<DatabaseConfig>,
-  pub(crate) rate_limit: Option<RateLimitConfig>,
-  pub(crate) federation: Option<FederationConfig>,
-  pub(crate) hostname: Option<String>,
-  pub(crate) bind: Option<IpAddr>,
-  pub(crate) port: Option<u16>,
-  pub(crate) tls_enabled: Option<bool>,
-  pub(crate) jwt_secret: Option<String>,
-  pub(crate) pictrs_url: Option<String>,
-  pub(crate) iframely_url: Option<String>,
-  pub(crate) captcha: Option<CaptchaConfig>,
-  pub(crate) email: Option<EmailConfig>,
-  pub(crate) setup: Option<SetupConfig>,
-  pub(crate) additional_slurs: Option<String>,
-  pub(crate) actor_name_max_length: Option<usize>,
+  #[serde(default)]
+  pub database: DatabaseConfig,
+  #[default(Some(RateLimitConfig::default()))]
+  pub rate_limit: Option<RateLimitConfig>,
+  #[default(FederationConfig::default())]
+  pub federation: FederationConfig,
+  #[default(CaptchaConfig::default())]
+  pub captcha: CaptchaConfig,
+  #[default(None)]
+  pub email: Option<EmailConfig>,
+  #[default(None)]
+  pub setup: Option<SetupConfig>,
+  #[default("unset")]
+  pub hostname: String,
+  #[default(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)))]
+  pub bind: IpAddr,
+  #[default(8536)]
+  pub port: u16,
+  #[default(true)]
+  pub tls_enabled: bool,
+  #[default("changeme")]
+  pub jwt_secret: String,
+  #[default(None)]
+  pub pictrs_url: Option<String>,
+  #[default(None)]
+  pub iframely_url: Option<String>,
+  #[default(None)]
+  pub additional_slurs: Option<String>,
+  #[default(20)]
+  pub actor_name_max_length: usize,
```

Comment:
I definitely like this better, as its more clear about what's optional and what isn't, but it doesn't seem like it works with merge.